src/Microsoft.Azure.ServiceBus/MessageBatchHandlerOptions.cs

```diff
@@ -0,0 +1,115 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.Azure.ServiceBus
+{
+    using System;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Primitives;
+    using System.Collections.Generic;
+
+    /// <summary>Provides options associated with message pump processing using
+    /// <see cref="QueueClient.RegisterMessageBatchHandler(Func{IList{Message}, CancellationToken, Task}, MessageBatchHandlerOptions)" /> and
+    /// <see cref="SubscriptionClient.RegisterMessageBatchHandler(Func{IList{Message}, CancellationToken, Task}, MessageBatchHandlerOptions)" />.</summary>
+    public sealed class MessageBatchHandlerOptions
```

Comment:
>MessageBatchHandlerOptions [](start = 24, length = 26)

This should just derive from the non-batch handler and add the extra property.

Reply:
Not a big fan of inheritance. A slight change will cause waves throughout all derived classes.